packages/terminal/src/widget.ts

```diff
@@ -65,65 +67,48 @@ export class Terminal extends Widget {
       this._session.messageReceived.disconnect(this._onMessage, this);
     }
     this._session = value || null;
-    if (!value) {
-      return;
+    if (value) {
+      this._setSession(value);
     }
-    value.ready.then(() => {
-      if (this.isDisposed || value !== this._session) {
-        return;
-      }
-      value.messageReceived.connect(
-        this._onMessage,
-        this
-      );
-      this.title.label = `Terminal ${value.name}`;
-      this._setSessionSize();
-      if (this._options.initialCommand) {
-        this._session.send({
-          type: 'stdin',
-          content: [this._options.initialCommand + '\r']
-        });
-      }
-    });
   }
 
   /**
-   * Set a config option for the terminal.
+   * Get a config option for the terminal.
    */
-  getOption<K extends keyof Terminal.IOptions>(
+  getOption<K extends keyof ITerminal.IOptions>(
     option: K
-  ): Terminal.IOptions[K] {
+  ): ITerminal.IOptions[K] {
     return this._options[option];
   }
 
   /**
    * Set a config option for the terminal.
    */
-  setOption<K extends keyof Terminal.IOptions>(
+  setOption<K extends keyof ITerminal.IOptions>(
     option: K,
-    value: Terminal.IOptions[K]
+    value: ITerminal.IOptions[K]
   ): void {
     if (this._options[option] === value) {
       return;
     }
 
     this._options[option] = value;
 
-    if (option === 'initialCommand') {
-      return;
-    }
-
-    if (option === 'theme') {
-      if (value === 'light') {
-        this.addClass('jp-mod-light');
-        this._term.setOption('theme', Private.lightTheme);
-      } else {
-        this.removeClass('jp-mod-light');
-        this._term.setOption('theme', Private.darkTheme);
-      }
-    } else {
-      this._term.setOption(option, value);
-      this._needsResize = true;
+    switch (option) {
+      case 'initialCommand':
+        return;
```

Comment:
should we add a deprecation warning, or just yank it altogether? of course, I did always love `initialCommand` for demos, cuz you could make an `a` with a `data-` tag and lab would happily launch an `rm -rf /` terminal...

Reply:
My position had been that `initialCommand` is okay if it is driven by extension code, but that we shouldn't allow it in workspaces. But you raise a good point -- here is another possible way to use it maliciously.

We've discussed refactoring the command linker to be a bit safer before, but that didn't really go anywhere...

I think you've just talked me into removing it all-together.

Reply:
awww, i guess that's why we can't have nice things...

Reply:
:(

Reply:
Renewed discussion in #6183